packages/runtime-core/src/components/Teleport.ts

```diff
@@ -248,7 +248,7 @@ export const TeleportImpl = {
             parentComponent,
             parentSuspense,
             true,
-            optimized
+            false
```

Comment:
Define the child as a variable

```js
const child = (children as VNode[])[i]
```

And then:

```suggestion
           !!child.dynamicChildren
```